app/queries/get_dependents.rb

```diff
@@ -0,0 +1,37 @@
+module Queries
+  class GetDependents
+    # Finds all content_ids that depend on the given content_id. This query is
+    # recursive and will also return content_ids that aren't direct neighbours
+    # to the given content_id. This is called a 'transitive closure'.
+    #
+    # If the optional link_type is given, only links that have that link_type
+    # will be followed when computing dependencies.
+    def call(content_id, link_type = nil)
+      results(content_id, link_type).map do |result|
+        result.fetch("content_id")
+      end
+    end
```

Comment:
I think we can achieve the same results by SELECTing multiple link types in the query itself. One query would be better than one for each link type if possible.